R/epinow.R

```diff
@@ -346,94 +349,128 @@ regional_epinow <- function(reported_cases,
-  
+
   if (missing(target_date)) {
     target_date <- as.character(max(reported_cases$date))
   }
-  
+
   if (missing(target_folder)) {
     target_folder <- NULL
   }
-  
+
   futile.logger::flog.info("Reporting estimates using data up to: %s", target_date)
-  
-  
+
+
   ## Check for regions more than required time points with cases
-  eval_regions <- data.table::copy(reported_cases)[,.(confirm = confirm > 0), by = c("region", "date")][,
-            .(confirm = sum(confirm, na.rm = TRUE)), by = "region"][confirm >= non_zero_points]$region
-  
+  eval_regions <- data.table::copy(reported_cases)[, .(confirm = confirm > 0), by = c("region", "date")][,
+    .(confirm = sum(confirm, na.rm = TRUE)), by = "region"][confirm >= non_zero_points]$region
+
   eval_regions <- unique(eval_regions)
-  
+
   ## Exclude zero regions
   reported_cases <- reported_cases[!is.na(region)][region %in% eval_regions]
-  
+
   futile.logger::flog.info("Producing estimates for: %s",
-          paste(eval_regions, collapse = ", "))
-  
+                           paste(eval_regions, collapse = ", "))
+
   ## regional pipelines
   regions <- unique(reported_cases$region)
-  
+
   ## Function to run the pipeline in a region
-  run_region <- function(target_region, 
+  run_region <- function(target_region,
                          reported_cases,
                          cores = cores,
-                         ...) { 
+                         return_timings = return_timings,
+                         max_execution_time = max_execution_time,
+                         ...) {
     futile.logger::flog.info("Initialising estimates for: %s", target_region)
-    
+
     data.table::setDTthreads(threads = 1)
 
     if (!is.null(target_folder)) {
       target_folder <- file.path(target_folder, target_region)
     }
-    
+
+    futile.logger::flog.trace("filtering data for target region %s", target_region)
     regional_cases <- reported_cases[region %in% target_region][, region := NULL]
-    
-    out <- EpiNow2::epinow(
-      reported_cases = regional_cases,
-      target_folder = target_folder,
-      target_date = target_date, 
-      return_estimates = TRUE,
-      cores = cores,
-      ...)
-    
-     futile.logger::flog.info("Completed estimates for: %s", target_region)
-
-     return(out)
+
+    futile.logger::flog.trace("calling epinow2::epinow to process data for %s", target_region)
+    timing <- system.time(
+      out <- tryCatch(
+        R.utils::withTimeout(
+          withCallingHandlers(EpiNow2::epinow(
+            reported_cases = regional_cases,
+            target_folder = target_folder,
+            target_date = target_date,
+            return_estimates = TRUE,
+            cores = cores,
+            ...),
+                              warning = function(w) {
+                                futile.logger::flog.warn("%s: %s - %s", target_region, w$message, toString(w$call))
+                                cnd_muffle(w)
```

Comment:
suggest call package to fit style (but not required)

Reply:
I'm not sure I understand, would you mind expanding?

Reply:
as in everywhere else we have used `purrr::map` so here it should really be `rlang::cnd_muffle` to match. That approaches has issues though as code can pass checks even when a function is missing from the NAMESPACE (like `purrr::keep` in this PR) so TBH may not be ideal practice.